modules/runner/src/integrationTest/java/org/apache/ignite/internal/sql/engine/ItDmlTest.java

```diff
@@ -350,6 +351,33 @@ public void testMergeKeysConflict() {
         assertEquals(Sql.DUPLICATE_KEYS_ERR, ex.code());
     }
 
+    /**
+     * Test verifies that scan is executed within provided transaction.
+     */
+    @Test
+    @Disabled("https://issues.apache.org/jira/browse/IGNITE-17309")
+    public void scanExecutedWithinGivenTransaction() {
+        sql("CREATE TABLE test (id int primary key, val int)");
+
+        Transaction tx = CLUSTER_NODES.get(0).transactions().begin();
+
+        sql(tx, "INSERT INTO test VALUES (0, 0)");
+
+        // just inserted row should be visible within the same transaction
+        assertEquals(1, sql(tx, "select * from test").size());
+
+        Transaction anotherTx = CLUSTER_NODES.get(0).transactions().begin();
```

Comment:
I suggest to add third tx which started before the first transaction and do the same checks.

Reply:
I believe the test should be minimal yet enough to verify the part it's supposed to verify. This particular test is aimed to verify that external transaction is used for scan operation.